### ISSUE TITLE
Image comparison of the ground truth depth and predicted depth

### ISSUE THREAD

**Issue by MeghaMaheshwari** [mrharicot/monodepth#30]
I generated images from the ground truth depth data and the predicted depth data based on the stereo_files_2015.txt files that contain the exact images that has been used by you. However, the predicted depth images differ significantly from the ground truth images and this is making it hard for me to understand if the predicted depths are really correct and can be used later. Could you please advise?

All I did was just save the images using scipy in evaluate_utils.py using the code

      gt_path = "monodepth/gt/" + str(i) + ".png"
       
        path = "monodepth/pred/"+ str(i) + ".png"
        scipy.misc.imsave(path, pred_depth)
        scipy.misc.imsave(gt_path,gt_depth)

![0_groundtruth](https://user-images.githubusercontent.com/8704236/28804377-9ff1c596-7617-11e7-9e58-21965f245cab.png)
![0_predicted](https://user-images.githubusercontent.com/8704236/28804380-a7b7c5be-7617-11e7-9101-396bed3fcbcd.png)

Please let me know if you need more information. These are the values i got on the 200 images

abs_rel,     sq_rel,        rms,    log_rms,     d1_all,         a1,         a2,         a3
    0.0897,     1.2037,      5.157,      0.166,     16.366,      0.915,      0.969,      0.986

**Comment by mrharicot** [comment]
Hi,

Your errors seem quite low, it looks like the training went ok.
I do not visualize the depth directly but the inverse of it (or disparity).

If you try to save 1.0 / pred_depth you should see a reasonable image.

**Comment by MeghaMaheshwari** [comment]
ok, let me try that

**Comment by MeghaMaheshwari** [comment]
ok, I got the disparity images which look reasonable. By the way in the function def scale_pyramid(self, img, num_scales): I had to cast the nh and nw as tensor elements otherwise the code was giving errors, 

 nh = tf.cast(nh, tf.int32)
 nw = tf.cast(nw, tf.int32)
 scaled_imgs.append(tf.image.resize_area(img, [nh, nw]))

The above change worked for me. I am using Python3 though and I guess your code is based on Python2.

I had one more question - your encoder for vgg as I see is pretty straightforward but for the decoder part is there any related article or link that I can read upon to get a better feel of it and how the disparities are getting output at different levels.
